source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -50,11 +53,111 @@ void Filter::onRead() {
 
 void Filter::onReadWorker() {
   Network::ConnectionSocket& socket = cb_->socket();
-  std::string proxy_line;
-  if (!readLine(socket.fd(), proxy_line)) {
+  WireHeader hdr;
```

Comment:
Sorry I don't think I was clear here. I meant to remove the valid_ member from `WireHeader`, and declare an `absl::optional<WireHeader>` here, and then pass that to `readProxyHeader` by reference. Then, that function can fill it if complete.